main.ts

```diff
@@ -109,9 +137,114 @@ async function run(): Promise<void> {
     })) {
       ln(`/dev/${linkPath}`, `/proc/self/${target}`)
     }
+
+    if (msystem === 'CLANGARM64') {
+      // ARM64 dependencies aren't included yet in the Git for Windows SDK. Ask Pacman to install them.
+      core.startGroup(`Installing CLANGARM64 dependencies`)
+      await installArm64Dependencies(outputDirectory)
+      core.endGroup()
+    }
   } catch (error) {
     core.setFailed(error instanceof Error ? error.message : `${error}`)
   }
 }
 
-run()
+async function installArm64Dependencies(
+  outputDirectory: string
+): Promise<void> {
+  mkdirp(`${outputDirectory}/clangarm64`)
+  fs.appendFileSync(
+    `${outputDirectory}/etc/pacman.conf`,
+    `
+    [clangarm64]
+    Server = https://mirror.msys2.org/mingw/clangarm64/`
+  )
+
+  const child = spawn('pacman.exe', [
+    '-Sy',
+    '--noconfirm',
+    'base-devel',
+    'mingw-w64-clang-aarch64-openssl',
+    'mingw-w64-clang-aarch64-zlib',
+    'mingw-w64-clang-aarch64-curl',
+    'mingw-w64-clang-aarch64-expat',
+    'mingw-w64-clang-aarch64-libiconv',
+    'mingw-w64-clang-aarch64-toolchain',
+    'mingw-w64-clang-aarch64-pcre2',
+    'mingw-w64-clang-aarch64-libssp'
+  ])
+
+  child.stdout.setEncoding('utf-8')
+  child.stderr.setEncoding('utf-8')
+
+  child.stdout.on('data', data => {
+    core.info(data)
+  })
+
+  child.stderr.on('data', data => {
+    core.error(data)
+  })
+
+  return new Promise((resolve, reject) => {
+    child.on('error', error => reject(error))
+    child.on('close', status =>
+      status === 0
+        ? resolve()
+        : reject(new Error(`Process exited with status code ${status}`))
+    )
+  })
+}
+
+function cleanup(): void {
+  const {artifactName} = getArtifactMetadata(flavor, architectureToDownload)
+  const outputDirectory = core.getInput('path') || `C:/${artifactName}`
+
+  /**
+   * Shelling out to rm -rf is more than twice as fast as Node's fs.rmSync method.
+   * Let's use it if it's available, and otherwise fall back to fs.rmSync.
+   */
+  const cleanupMethod = fs.existsSync(`${gitForWindowsUsrBinPath}/bash.exe`)
+    ? 'native'
+    : 'node'
+
+  core.info(
+    `Cleaning up ${outputDirectory} using the "${cleanupMethod}" method...`
+  )
+
+  if (cleanupMethod === 'native') {
+    const child = spawnSync(
+      `${gitForWindowsUsrBinPath}/bash.exe`,
+      ['-c', `rm -rf "${outputDirectory}"`],
+      {
+        encoding: 'utf-8',
+        env: {PATH: '/usr/bin'}
+      }
+    )
+
+    if (child.error) throw child.error
+    if (child.stderr) core.error(child.stderr)
+  } else {
+    fs.rmSync(outputDirectory, {recursive: true, force: true})
+  }
+
+  core.info(`Finished cleaning up ${outputDirectory}.`)
+}
+
+/**
+ * Indicates whether the POST action is running
+ */
+export const isPost = !!process.env['STATE_isPost']
+
+// Publish a variable so that when the POST action runs, it can determine it should run the cleanup logic.
```

Comment:
Hmm. Maybe we can separate out the `cleanup` logic into its own commit, and then figure out how to detect self-hosted runners and only spend the time to clean up on those?

Reply:
Following up in https://github.com/git-for-windows/setup-git-for-windows-sdk/pull/485